NotificationHubs/src/com/windowsazure/messaging/WindowsCredential.java

```diff
@@ -1,54 +1,90 @@
 import java.util.ArrayList;
 import java.util.List;
 
-public final class WindowsCredential extends PnsCredential {	
-	private String packageSid;
-	private String secretKey;
-	
-	public WindowsCredential(){
-		this(null,null);
-	}
-		
-	public WindowsCredential(String packageSid, String secretKey){
-		super();
-		this.setPackageSid(packageSid);
-		this.setSecretKey(secretKey);
-	}
-	
-	public String getPackageSid() {
-		return packageSid;
-	}
-
-	public void setPackageSid(String packageSid) {
-		this.packageSid = packageSid;
-	}
-
-	public String getSecretKey() {
-		return secretKey;
-	}
-
-	public void setSecretKey(String secretKey) {
-		this.secretKey = secretKey;
-	}	
-	
-    	public void setWindowsLiveEndpoint(String propertyValue) throws Exception {
-        	// fix for reflection that's calling 'setWindowsLiveEndpoint' of null.
-        	// unused function
-    	}
-
-	@Override
-	public List<SimpleEntry<String, String>> getProperties() {
-		ArrayList<SimpleEntry<String, String>> result = new ArrayList<SimpleEntry<String, String>>();
-		result.add(new SimpleEntry<String, String>("PackageSid",getPackageSid()));
-		result.add(new SimpleEntry<String, String>("SecretKey",getSecretKey()));
-		return result;
-	}
-
-	@Override
-	public String getRootTagName() {
-		return "WnsCredential";
-	}
+/**
+ * This class represents a WNS credential.
+ */
+public final class WindowsCredential extends PnsCredential {
+    private String packageSid;
+    private String secretKey;
+
+    /**
+     * Creates a WNS credential.
+     */
+    public WindowsCredential() {
+        this(null, null);
+    }
+
+    /**
+     * Creates a WNS credential with package SDI and secret key.
+     *
+     * @param packageSid The package SID for the WNS credentials.
+     * @param secretKey  The secret key for the WNS credentials.
+     */
+    public WindowsCredential(String packageSid, String secretKey) {
+        super();
+        this.setPackageSid(packageSid);
+        this.setSecretKey(secretKey);
+    }
+
+    /**
+     * Gets the package SID for the WNS credentials.
+     *
+     * @return The package SID for the WNS credentials.
+     */
+    public String getPackageSid() {
+        return packageSid;
+    }
+
+    /**
+     * Sets the package SDI for the WNS credentials.
+     *
+     * @param packageSid The package SDI for the WNS credentials.
+     */
+    public void setPackageSid(String packageSid) {
+        this.packageSid = packageSid;
+    }
+
+    /**
+     * Gets the secret key for the WNS credentials.
+     *
+     * @return The secret key for the WNS credentials.
+     */
+    public String getSecretKey() {
+        return secretKey;
+    }
+
+    /**
+     * Sets the secret key for the WNS credentials.
+     *
+     * @param secretKey The secret key for the WNS credentials.
+     */
+    public void setSecretKey(String secretKey) {
+        this.secretKey = secretKey;
+    }
+
+    public void setWindowsLiveEndpoint(String propertyValue) {
```

Comment:
Does removing the checked exception cause a binary compat issue?

Reply:
It was never thrown in the first place